vendor/bat-native-ads/src/bat/ads/internal/resources/behavioral/purchase_intent/purchase_intent_signal_history_info.cc

```diff
@@ -68,35 +39,28 @@ bool PurchaseIntentSignalHistoryInfo::operator!=(
   return !(*this == rhs);
 }
 
-std::string PurchaseIntentSignalHistoryInfo::ToJson() const {
-  base::Value dictionary(base::Value::Type::DICTIONARY);
-
-  dictionary.SetStringKey("timestamp_in_seconds",
-                          base::NumberToString(created_at.ToDoubleT()));
-
-  dictionary.SetIntKey("weight", static_cast<int>(weight));
-
-  std::string json;
-  base::JSONWriter::Write(dictionary, &json);
+base::Value::Dict PurchaseIntentSignalHistoryInfo::ToValue() const {
+  base::Value::Dict dict;
+  dict.Set("timestamp_in_seconds",
+           base::NumberToString(created_at.ToDoubleT()));
 
-  return json;
+  dict.Set("weight", static_cast<int>(weight));
+  return dict;
 }
 
-bool PurchaseIntentSignalHistoryInfo::FromJson(const std::string& json) {
-  absl::optional<base::Value> value = base::JSONReader::Read(json);
-  if (!value) {
-    return false;
+bool PurchaseIntentSignalHistoryInfo::FromValue(const base::Value::Dict& root) {
+  weight = static_cast<uint16_t>(root.FindInt("weight").value_or(0));
+
+  if (const auto* value = root.FindString("timestamp_in_seconds")) {
+    double value_as_double = 0;
+    if (base::StringToDouble(*value, &value_as_double))
+      created_at = base::Time::FromDoubleT(value_as_double);
+    else
+      created_at = base::Time();
```

Comment:
nit: for consistency in ads can we please wrap with `{ } else { }`

Reply:
Done in fbad64b1176fd9700960386b467bc7d332c4af7e.